pyeudiw/jwt/__init__.py

```diff
@@ -38,23 +46,51 @@
     "EC": "A256GCM"
 }
 
+KeyLike = ECKey | RSAKey | OKPKey | SYMKey
+SerializationFormat = Literal["compact", "json"]
 
-class JWEHelper():
-    """
-    The helper class for work with JWEs.
-    """
 
-    def __init__(self, jwk: Union[JWK, dict]):
+logger = logging.getLogger(__name__)
+
+
+class JWHelperInterface:
+    def __init__(self, jwks: list[KeyLike | dict] | KeyLike | dict):
         """
         Creates an instance of JWEHelper.
 
-        :param jwk: The JWK used to crypt and encrypt the content of JWE.
-        :type jwk: JWK
+        :param jwks: The list of JWK used to crypt and encrypt the content of JWE.
+
         """
-        self.jwk = jwk
-        if isinstance(jwk, dict):
-            self.jwk = JWK(jwk)
-        self.alg = DEFAULT_SIG_KTY_MAP[self.jwk.key.kty]
+        if isinstance(jwks, dict):
+            single_jwk = key_from_jwk_dict(jwks)
+            single_jwk.add_kid()
+            self.jwks = [single_jwk]
+        elif isinstance(jwks, list):
+            self.jwks = []
+            for j in jwks:
+                if isinstance(j, dict):
+                    j = key_from_jwk_dict(j)
+                j.add_kid()
+                self.jwks.append(j)
+        elif isinstance(jwks, (ECKey, RSAKey, OKPKey, SYMKey)):
+            jwks.add_kid()
+            self.jwks = [jwks]
+        else:
+            logger.warning(f"Unhandled type {type(jwks)} for jwks")
```

Comment:
If the function receives as input a wrong unhandled type, than it should raise a `TypeError`, not log a warning and then possibly enter into an invalid state.
For example, if `self.jwks` is empty (which is the case of line 80), operations below `encrypt` and `sign` silently fails (in case of encrypt no exception is raised and an empty array is returned).

IMO TypeErrors in 99% of the cases are developers errors and should be treated as unrecoverable exceptions.

Reply:
I agree that raising a TypeError could be more appropriate in scenarios where the provided input is outright invalid and unrecoverable.

Reply:
aggiungere il raise dell'eccezione pertanto